src/app/clusters/level-control/level-control.cpp

```diff
@@ -1425,22 +1436,55 @@ void emberAfOnOffClusterLevelControlEffectCallback(EndpointId endpoint, bool new
     transitionTime.SetNull();
 #endif // IGNORE_LEVEL_CONTROL_CLUSTER_ON_OFF_TRANSITION_TIME
 
+    // Check if we're already performing an on/off transition.
+    if (state->commandIsOnOff)
```

Comment:
It may be possible to introduce some state instead of comparing the `moveToLevel` against the limits?